core/reactor.c

```diff
@@ -51,6 +51,26 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 // Global variable defined in tag.c:
 extern instant_t start_time;
 
+int lf_thread_id() {
+    return 0;
```

Comment:
These functions look sus. Why are they needed?

Reply:
What is sus about them? I am trying to avoid race conditions by letting different threads access their own dedicated data structures according to their thread ID. In a single-threaded context, we can simply say that the thread ID is always 0.

Reply:
I was missing the bit that this only applies to single threaded. Why is that though? There is no obvious preprocessor check here. Where is the implementation for threaded?

Reply:
All of `core/reactor.c` is for single-threaded and `core/threaded/reactor-threaded` is the counterpart for threaded. To mitigate duplication we have `core/reactor_common.c` which is used in both cases.